### PR TITLE
Build glib on x86-linux

### PR DESCRIPTION
**Pull request by koprok**
Changes to build glib on x86-linux target. The following triplet file has been used for the build:
```
set(VCPKG_TARGET_ARCHITECTURE x86)
set(VCPKG_CRT_LINKAGE dynamic)
set(VCPKG_LIBRARY_LINKAGE static)

set(VCPKG_CMAKE_SYSTEM_NAME Linux)

set(VCPKG_C_FLAGS "-m32")
set(VCPKG_CXX_FLAGS "-m32")
set(VCPKG_LINKER_FLAGS "-m32")
```

### CODE REVIEW

**Comment by ras0219-msft** [comment]
Thank for the PR! LGTM pending CI.

We generally try to make the CMake files as vcpkg-independent as possible, so I've pulled some of the CFLAGS detection logic over from our boost wrapper[1] which should be more robust :)

[1] https://github.com/Microsoft/vcpkg/blob/6509199a2aeb882c91babb7cc95ef36b30738091/ports/boost-modular-build-helper/CMakeLists.txt#L44-L60

File: ports/libffi/CMakeLists.txt [hunk 1]
```diff
@@ -97,30 +97,30 @@ macro(add_assembly ASMFILE)
     endif()
 endmacro()
 
-if("${TARGET}" STREQUAL "X86")
+if(TARGET STREQUAL "X86")
```
Comment (vicroms) [inline ports/libffi/CMakeLists.txt]: Doesn't work on CI nor locally for me without the `${}`.
Is this a CMake version issue?

Comment (koprok) [reply ports/libffi/CMakeLists.txt]: Hello @vicroms, @ras0219-msft 

The actual fix is commit ba13dc1. The commit that introduces the `${TARGET} -> TARGET` change is only a code cleanup.

Now I have tried to build the branch on my system and indeed it does not build. The error is:
```
CMake Error at CMakeLists.txt:100 (if):
  if given arguments:

    "TARGET" "STREQUAL" "X86"

  Unknown arguments specified
```

@ras0219-msft , should we revert the `${TARGET} -> TARGET` change?

Comment (vicroms) [reply ports/libffi/CMakeLists.txt]: According to the [CMake docs](https://cmake.org/cmake/help/latest/command/if.html) `if (VARIABLE)` should work. I'm not sure if `TARGET` has a specific meaning in that context, that stops it from working.

Given that using `${TARGET}` builds successfully, just reverting the changes is OK.

**Comment by vicroms** [comment]
Thanks for the PR! 
It passes all CI tests! 🎉

**Pull request merged**